oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,210 @@
+
+To alleviate these issues, we will switch from using the edx-transifex-bot to the GitHub
+Transifex App, a stable app provided by Transifex. Benefits of this change include being
+easier to maintain and solving a lot of the pain points. As part of this proposal, we
+suggest moving translations into their own repository, to make using  the GitHub
+Transifex app more streamlined and straightforward, and in order to make organizing and
+using the up to date translations simpler.
+
+Current State
+*************
+
+* Edx-transifex-bot is a potential security issue: The edx-transifex-bot requires admin
+  rights on Transifex in order to function. Admin rights give access to private/sensitive
+  information as well as the ability to permanently delete translation and configuration
+  files. At some point, the login to the edx-transifex-bot user was lost, and without
+  access to the scripts that the bot uses to function, this edx-transifex-bot is a
+  security issue we cannot control or debug.
+* Edx-transifex-bot is a black box: Login/scripts for edx-transifex-bot cannot be found
+  on GitHub or on wiki/secrets pages, and it is difficult to observe what work it is
+  supposed to do, or whether it is doing it correctly.
+* The translations for the Open edX Maple Release were never uploaded to Transifex,
+  because the automation handled by the edx-transifex-bot never uploaded it.
+* The translations for the Open edX Nutmeg Release were nearly not uploaded because the
+  edx-transifex-bot did not automatically upload it. In addition, the script for
+  uploading the translations did not work, and the script had to be tinkered for the new
+  Transifex API in order to upload it in time for release. The files uploaded were still
+  not in the preferred format because the new version of the Transifex API does not have
+  all the same features as the previous API version.
+
+  * Running the script to upload translations requires Transifex admin rights, and there
+    are few people with Admin rights to Transifex and knowledge of the Transifex API. We
+    should use the solution provided by Transifex to solve this type of problem.
+
+Rationale
+*********
+
+* This is an upgrade of a system we use regularly, but do not want to have to maintain
+  regularly.
+* Upgrading from a bot (machine user) to an app/workflow is recommended by GitHub and
+  makes the translation process more open source. 
+* The GitHub Transifex Integration App is developed and supported by Transifex
+* The Github Transifex App is very simple to configure and with many options. We can set
+  Transifex Projects to automatically upload/download translation files from a repository
+  once the translations are reviewed and accepted.
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower cost: we pay by the project so we end up paying less, and by
```

Comment:
We do pay for Transifex, though we have a special agreement as a non-profit/open-source project.  That said, I'm not sure about whether this proposal would cost less or whether cost should be a first class consideration in our architecture for translations.

Reply:
Agree re: slack that monetary cost should be dropped as a consideration

Reply:
Point out the human cost of managing the translation files from many repos on Transifex and remove the monetary capital. "We estimate that it takes X hours per month to manage this, and this will reduce it by Y%."